app/assets/javascripts/discourse/app/templates/discovery.hbs

```diff
@@ -3,6 +3,7 @@
 {{else}}
   <div class="container">
     {{discourse-banner user=currentUser banner=site.banner}}
+    {{category-read-only-banner category=category readOnly=navigationCategory.cannotCreateTopicOnCategory user=currentUser}}
```

Comment:
You probably don't need to pass `currentUser` in here, as all components have access to `this.currentUser`.